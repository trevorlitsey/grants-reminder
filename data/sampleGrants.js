import moment from 'moment';

const blankGrant = {
	id: 'blankGrant',
	date: moment(),
	name: '',
	purpose: '',
	notes: '',
	reminders: {
		twoMonths: false,
		oneMonth: false,
		twoWeeks: false,
		oneWeek: false,
	},
}

const basicGrant = {
	id: 'basicGrant',
	date: 1521592644420,
	name: 'a grant name',
	notes: '',
	purpose: 'general operations',
	reminders: {
		twoMonths: true,
		oneMonth: false,
		twoWeeks: true,
		oneWeek: false,
	},
}

const twoMonths = {
	id: 'twoMonths',
	name: 'NEA',
	date: 1525669200000,
	notes: '',
	purpose: 'general operations',
	reminders: {
		twoMonths: true,
		oneMonth: false,
		twoWeeks: false,
		oneWeek: false,
	}
}

const oneMonth = {
	id: 'oneMonth',
	notes: '',
	purpose: 'general operations',
	name: 'Ditmas',
	date: 1523077200000,
	reminders: {
		twoMonths: false,
		oneMonth: true,
		twoWeeks: false,
		oneWeek: true,
	}
}

const twoWeeks = {
	id: 'twoWeeks',
	notes: '',
	purpose: 'general operations',
	name: 'Aaron Copland',
	date: 1521608400000,
	reminders: {
		twoMonths: false,
		oneMonth: false,
		twoWeeks: true,
		oneWeek: false,
	}
}

const oneWeek = {
	id: 'oneWeek',
	notes: '',
	purpose: 'general operations',
	name: 'Koussevitzky',
	date: 1521003600000,
	reminders: {
		twoMonths: false,
		oneMonth: false,
		twoWeeks: false,
		oneWeek: true,
	}
}

const thisDay = {
	id: 'thisDay',
	notes: '',
	purpose: 'general operations',
	name: 'NMUSA',
	date: 1520402400000,
	reminders: {
		twoMonths: false,
		oneMonth: false,
		twoWeeks: false,
		oneWeek: false,
	}
}

const noDay = {
	id: 'noDay',
	notes: '',
	purpose: 'general operations',
	name: 'decoy',
	date: 1576452400000,
	reminders: {
		twoMonths: true,
		oneMonth: true,
		twoWeeks: true,
		oneWeek: true,
	}
}

const sampleGrants = {
	blankGrant,
	basicGrant,
	twoMonths,
	oneMonth,
	twoWeeks,
	oneWeek,
	thisDay,
	noDay
}

module.exports = sampleGrants;