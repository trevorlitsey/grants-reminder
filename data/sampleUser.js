const sampleGrants = require('./sampleGrants');

const { twoMonths, oneMonth } = sampleGrants

const sampleUser = {
	'email': 'trevorlitsey@gmail.com',
	'purposes': {
		'one': 'purpose1!',
		'two': 'purpose2!',
		'three': 'purpose3!',
	},
	'grants': {
		oneMonth,
		twoMonths,
		'jeesc9a9': {
			'date': 1523030464444,
			'id': 'jeesc9a9',
			'name': 'a grant due in june!',
			'notes': 'so many notes',
			'purpose': 'General Operations',
			'reminders': {
				'oneMonth': false,
				'oneWeek': true,
				'twoMonths': false,
				'twoWeeks': false
			}
		},
		'jeesm1xk': {
			'date': 1552338664444,
			'id': 'jeesm1xk',
			'name': 'yep',
			'notes': 'notes!!',
			'purpose': 'Education',
			'reminders': {
				'oneMonth': false,
				'oneWeek': false,
				'twoMonths': true,
				'twoWeeks': false
			}
		},
		'jek5wems': {
			'date': 1662737764444,
			'id': 'jek5wems',
			'name': 'a great name',
			'notes': 'these are incredible notes!',
			'purpose': 'General Operations',
			'reminders': {
				'oneMonth': false,
				'oneWeek': false,
				'twoMonths': false,
				'twoWeeks': false
			}
		}
	}
}

module.exports = sampleUser;
