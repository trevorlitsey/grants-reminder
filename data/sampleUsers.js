const sampleUsers = {
  "test": {
    "jeei5oh5": {
      "date": "2018/2/5",
      "id": "jeei5oh5",
      "name": "A Grant",
      "notes": "these are some notes!",
      "purpose": "Education",
      "reminders": {
        "oneMonth": true,
        "oneWeek": false,
        "twoMonths": true,
        "twoWeeks": false
      }
    },
    "jeek5k21": {
      "date": "2019/2/5",
      "id": "jeek5k21",
      "name": "Another grant",
      "notes": "some very good notes",
      "purpose": "Education",
      "reminders": {
        "oneMonth": true,
        "oneWeek": false,
        "twoMonths": false,
        "twoWeeks": true
      }
    }
  },
  "uCNiPR77B7UxHIFfrvyyl7wjsRl1": {
    "email": "trevorlitsey@gmail.com",
    "grants": {
      "jem0c85e": {
        "date": 1520724885671,
        "id": "jem0c85e",
        "name": "yp",
        "notes": "",
        "purpose": "",
        "reminders": {
          "oneMonth": false,
          "oneWeek": false,
          "twoMonths": false,
          "twoWeeks": false
        }
      }
    },
    "jees8no0": {
      "date": 1526227264444,
      "id": "jees8no0",
      "name": "A VERY big grants",
      "notes": "these are notes!!!!!",
      "purpose": "Another thing",
      "reminders": {
        "oneMonth": true,
        "oneWeek": false,
        "twoMonths": true,
        "twoWeeks": false
      }
    },
    "jeesc9a9": {
      "date": 1523030464444,
      "id": "jeesc9a9",
      "name": "a grant due in june!",
      "notes": "so many notes",
      "purpose": "General Operations",
      "reminders": {
        "oneMonth": false,
        "oneWeek": true,
        "twoMonths": false,
        "twoWeeks": false
      }
    },
    "jeesm1xk": {
      "date": 1552338664444,
      "id": "jeesm1xk",
      "name": "yep",
      "notes": "notes!!",
      "purpose": "Education",
      "reminders": {
        "oneMonth": false,
        "oneWeek": false,
        "twoMonths": true,
        "twoWeeks": false
      }
    },
    "jek5wems": {
      "date": 1662737764444,
      "id": "jek5wems",
      "name": "a great name",
      "notes": "these are incredible notes!",
      "purpose": "General Operations",
      "reminders": {
        "oneMonth": false,
        "oneWeek": false,
        "twoMonths": false,
        "twoWeeks": false
      }
    }
  }
}

module.exports = sampleUsers;
