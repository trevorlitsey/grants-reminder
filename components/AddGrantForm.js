import { object, func } from 'prop-types';
import moment from 'moment';

import handleGrantFormChange from './helpers/handleGrantFormChange';

import MonthOptions from './MonthOptions';
import DayOptions from './DayOptions';
import YearOptions from './YearOptions';
import PurposeOptions from './PurposeOptions';
import ReminderOptions from './ReminderOptions';
import FormAlert from './FormAlert';

import { blankGrant } from '../data/sampleGrants';

class AddGrantForm extends React.PureComponent {

	constructor() {
		super();

		this.handleGrantFormChange = handleGrantFormChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
		this.displayAlert = this.displayAlert.bind(this);

		this.state = {
			grant: {
				...blankGrant,
			},
			alertMessage: '',
		}
	}

	displayAlert(alertMessage) {
		this.setState({ alertMessage });
		setTimeout(() => {
			this.setState({ alertMessage: '' });
		}, 4000)
	}

	handleSubmit(e) {
		e.preventDefault();

		const { date, name, purpose, notes, reminders } = { ...this.state.grant };

		if (!name) return this.displayAlert('name of grant is required');

		this.props.addNewGrant({
			date: date.valueOf(),
			name,
			purpose,
			notes,
			reminders
		});

		// reset
		this.setState({ grant: { ...blankGrant } });
	}

	render() {

		const { name, date, purpose, notes, reminders } = this.state.grant;

		return (
			<div>
				<form onSubmit={this.handleSubmit}>
					<h5>Add New Grant: </h5>
					<div className="row">
						<input
							className="u-full-width four columns"
							type="text"
							placeholder="Name of grant"
							name="name"
							value={name}
							onChange={this.handleGrantFormChange}
							required
						/>
						<MonthOptions
							selected={date.month()}
							handleChange={this.handleGrantFormChange}
						/>
						<DayOptions
							date={date}
							selected={date.date()}
							handleChange={this.handleGrantFormChange}
						/>
						<YearOptions
							selected={date.year()}
							handleChange={this.handleGrantFormChange}
						/>
						<PurposeOptions
							purposes={this.props.purposes}
							selected={purpose}
							handleChange={this.handleGrantFormChange}
						/>
					</div>
					<textarea
						className="u-full-width"
						placeholder="Notes ..."
						name="notes"
						value={notes}
						onChange={this.handleGrantFormChange}
					></textarea>
					<ReminderOptions
						{...reminders}
						handleChange={this.handleGrantFormChange}
					/>
					<input
						className="button-primary"
						type="submit"
						value="Submit"
						data-test="purpose"
					/>
					<FormAlert message={this.state.alertMessage} />
				</form>

				<style jsx>{`
					.row {
							margin: 12px 0;
					}
			
					.options {
						display: grid;
						grid-template-columns: repeat(auto-fill, minmax(100px, 1fr));
					}

					.options label {
						margin: 0;
					}

					.button-primary {
						margin-top: 8px; 
					}
				
			`}</style>

			</div>
		)
	}
}

AddGrantForm.propTypes = {
	purposes: object.isRequired,
	addNewGrant: func.isRequired,
}

export default AddGrantForm;