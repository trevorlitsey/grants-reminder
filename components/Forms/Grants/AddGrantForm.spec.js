import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import { blankGrant, basicGrant } from '../../../data/sampleGrants';

import AddGrantForm from './AddGrantForm';

const fakeEvent = {
	preventDefault: () => { },
}

const expectedSubmittedGrant = {
	name: basicGrant.name,
	purpose: basicGrant.purpose,
	notes: basicGrant.notes,
	reminders: basicGrant.reminders,
	date: basicGrant.date.valueOf(),
}

describe('AddGrantForm', () => {

	it('should render without crashing', () => {
		const wrapper = shallowrenderAddGrantForm();
		const grant = {
			...basicGrant,
			date: moment('2018-03-23')
		}
		wrapper.setState({ grant });
		expect(wrapper).toMatchSnapshot();
	})

	it('should call addNewGrant on submit', () => {
		const grant = {
			...basicGrant,
			date: moment(basicGrant.date),
		}
		const instance = shallowrenderAddGrantForm().instance();
		instance.setState({ grant })
		instance.handleSubmit(fakeEvent)
		expect(instance.props.addNewGrant.mock.calls.length).toBe(1);
		expect(instance.props.addNewGrant.mock.calls[0][0]).toEqual(expectedSubmittedGrant);
	})

	it('should reset state on successful submit', () => {
		const grantState = {
			...basicGrant,
			date: moment(basicGrant.date),
		}
		const instance = shallowrenderAddGrantForm().instance();
		instance.setState({ grant: grantState })
		instance.handleSubmit(fakeEvent)
		expect(instance.state.grant).toEqual(blankGrant);
	})

	it('should display an alert if no grant name provided', () => {
		const e = {
			preventDefault: () => { },
		}
		const instance = shallowrenderAddGrantForm().instance();
		instance.handleSubmit(fakeEvent);
		expect(instance.state.alertMessage).toEqual('name of grant is required');
	})

})

function shallowrenderAddGrantForm(props = {}) {
	const propsToUser = {
		purposes: {
			one: 'General Operations',
			two: 'Education',
			three: 'Special Projects',
		},
		addNewGrant: jest.fn(),
		...props
	}
	return shallow(<AddGrantForm {...propsToUser} />)
}