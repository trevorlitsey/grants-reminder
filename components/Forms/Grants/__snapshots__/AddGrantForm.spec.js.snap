// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddGrantForm should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddGrantForm
    addNewGrant={[MockFunction]}
    purposes={
      Object {
        "one": "General Operations",
        "three": "Special Projects",
        "two": "Education",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="jsx-2962072046"
          onSubmit={[Function]}
        >
          <h5
            className="jsx-2962072046"
          >
            Add New Grant: 
          </h5>
          <div
            className="jsx-2962072046 row"
          >
            <input
              className="jsx-2962072046 u-full-width four columns"
              name="name"
              onChange={[Function]}
              placeholder="Name of grant"
              required={true}
              type="text"
              value="a grant name"
            />
            <MonthOptions
              handleChange={[Function]}
              selected={2}
            />
            <DayOptions
              date={"2018-03-23T05:00:00.000Z"}
              handleChange={[Function]}
              selected={23}
            />
            <YearOptions
              handleChange={[Function]}
              selected={2018}
            />
            <PurposeOptions
              handleChange={[Function]}
              purposes={
                Object {
                  "one": "General Operations",
                  "three": "Special Projects",
                  "two": "Education",
                }
              }
              selected="general operations"
            />
          </div>
          <textarea
            className="jsx-2962072046 u-full-width"
            name="notes"
            onChange={[Function]}
            placeholder="Notes ..."
            value=""
          />
          <ReminderOptions
            handleChange={[Function]}
            oneMonth={false}
            oneWeek={false}
            twoMonths={true}
            twoWeeks={true}
          />
          <input
            className="jsx-2962072046 button-primary"
            data-test="purpose"
            type="submit"
            value="Submit"
          />
          <FormAlert
            message=""
          />
        </form>,
        <JSXStyle
          css=".row.jsx-2962072046{margin:12px 0;}.options.jsx-2962072046{display:grid;grid-template-columns:repeat(auto-fill,minmax(100px,1fr));}textarea.jsx-2962072046{resize:vertical;}.options.jsx-2962072046 label.jsx-2962072046{margin:0;}.button-primary.jsx-2962072046{margin-top:8px;}"
          styleId="2962072046"
        />,
      ],
      "className": "jsx-2962072046",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="jsx-2962072046"
            >
              Add New Grant: 
            </h5>,
            <div
              className="jsx-2962072046 row"
            >
              <input
                className="jsx-2962072046 u-full-width four columns"
                name="name"
                onChange={[Function]}
                placeholder="Name of grant"
                required={true}
                type="text"
                value="a grant name"
              />
              <MonthOptions
                handleChange={[Function]}
                selected={2}
              />
              <DayOptions
                date={"2018-03-23T05:00:00.000Z"}
                handleChange={[Function]}
                selected={23}
              />
              <YearOptions
                handleChange={[Function]}
                selected={2018}
              />
              <PurposeOptions
                handleChange={[Function]}
                purposes={
                  Object {
                    "one": "General Operations",
                    "three": "Special Projects",
                    "two": "Education",
                  }
                }
                selected="general operations"
              />
            </div>,
            <textarea
              className="jsx-2962072046 u-full-width"
              name="notes"
              onChange={[Function]}
              placeholder="Notes ..."
              value=""
            />,
            <ReminderOptions
              handleChange={[Function]}
              oneMonth={false}
              oneWeek={false}
              twoMonths={true}
              twoWeeks={true}
            />,
            <input
              className="jsx-2962072046 button-primary"
              data-test="purpose"
              type="submit"
              value="Submit"
            />,
            <FormAlert
              message=""
            />,
          ],
          "className": "jsx-2962072046",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add New Grant: ",
              "className": "jsx-2962072046",
            },
            "ref": null,
            "rendered": "Add New Grant: ",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="jsx-2962072046 u-full-width four columns"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name of grant"
                  required={true}
                  type="text"
                  value="a grant name"
                />,
                <MonthOptions
                  handleChange={[Function]}
                  selected={2}
                />,
                <DayOptions
                  date={"2018-03-23T05:00:00.000Z"}
                  handleChange={[Function]}
                  selected={23}
                />,
                <YearOptions
                  handleChange={[Function]}
                  selected={2018}
                />,
                <PurposeOptions
                  handleChange={[Function]}
                  purposes={
                    Object {
                      "one": "General Operations",
                      "three": "Special Projects",
                      "two": "Education",
                    }
                  }
                  selected="general operations"
                />,
              ],
              "className": "jsx-2962072046 row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "jsx-2962072046 u-full-width four columns",
                  "name": "name",
                  "onChange": [Function],
                  "placeholder": "Name of grant",
                  "required": true,
                  "type": "text",
                  "value": "a grant name",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleChange": [Function],
                  "selected": 2,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "date": "2018-03-23T05:00:00.000Z",
                  "handleChange": [Function],
                  "selected": 23,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleChange": [Function],
                  "selected": 2018,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleChange": [Function],
                  "purposes": Object {
                    "one": "General Operations",
                    "three": "Special Projects",
                    "two": "Education",
                  },
                  "selected": "general operations",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "jsx-2962072046 u-full-width",
              "name": "notes",
              "onChange": [Function],
              "placeholder": "Notes ...",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleChange": [Function],
              "oneMonth": false,
              "oneWeek": false,
              "twoMonths": true,
              "twoWeeks": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "jsx-2962072046 button-primary",
              "data-test": "purpose",
              "type": "submit",
              "value": "Submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "message": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": ".row.jsx-2962072046{margin:12px 0;}.options.jsx-2962072046{display:grid;grid-template-columns:repeat(auto-fill,minmax(100px,1fr));}textarea.jsx-2962072046{resize:vertical;}.options.jsx-2962072046 label.jsx-2962072046{margin:0;}.button-primary.jsx-2962072046{margin-top:8px;}",
          "styleId": "2962072046",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="jsx-2962072046"
            onSubmit={[Function]}
          >
            <h5
              className="jsx-2962072046"
            >
              Add New Grant: 
            </h5>
            <div
              className="jsx-2962072046 row"
            >
              <input
                className="jsx-2962072046 u-full-width four columns"
                name="name"
                onChange={[Function]}
                placeholder="Name of grant"
                required={true}
                type="text"
                value="a grant name"
              />
              <MonthOptions
                handleChange={[Function]}
                selected={2}
              />
              <DayOptions
                date={"2018-03-23T05:00:00.000Z"}
                handleChange={[Function]}
                selected={23}
              />
              <YearOptions
                handleChange={[Function]}
                selected={2018}
              />
              <PurposeOptions
                handleChange={[Function]}
                purposes={
                  Object {
                    "one": "General Operations",
                    "three": "Special Projects",
                    "two": "Education",
                  }
                }
                selected="general operations"
              />
            </div>
            <textarea
              className="jsx-2962072046 u-full-width"
              name="notes"
              onChange={[Function]}
              placeholder="Notes ..."
              value=""
            />
            <ReminderOptions
              handleChange={[Function]}
              oneMonth={false}
              oneWeek={false}
              twoMonths={true}
              twoWeeks={true}
            />
            <input
              className="jsx-2962072046 button-primary"
              data-test="purpose"
              type="submit"
              value="Submit"
            />
            <FormAlert
              message=""
            />
          </form>,
          <JSXStyle
            css=".row.jsx-2962072046{margin:12px 0;}.options.jsx-2962072046{display:grid;grid-template-columns:repeat(auto-fill,minmax(100px,1fr));}textarea.jsx-2962072046{resize:vertical;}.options.jsx-2962072046 label.jsx-2962072046{margin:0;}.button-primary.jsx-2962072046{margin-top:8px;}"
            styleId="2962072046"
          />,
        ],
        "className": "jsx-2962072046",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5
                className="jsx-2962072046"
              >
                Add New Grant: 
              </h5>,
              <div
                className="jsx-2962072046 row"
              >
                <input
                  className="jsx-2962072046 u-full-width four columns"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name of grant"
                  required={true}
                  type="text"
                  value="a grant name"
                />
                <MonthOptions
                  handleChange={[Function]}
                  selected={2}
                />
                <DayOptions
                  date={"2018-03-23T05:00:00.000Z"}
                  handleChange={[Function]}
                  selected={23}
                />
                <YearOptions
                  handleChange={[Function]}
                  selected={2018}
                />
                <PurposeOptions
                  handleChange={[Function]}
                  purposes={
                    Object {
                      "one": "General Operations",
                      "three": "Special Projects",
                      "two": "Education",
                    }
                  }
                  selected="general operations"
                />
              </div>,
              <textarea
                className="jsx-2962072046 u-full-width"
                name="notes"
                onChange={[Function]}
                placeholder="Notes ..."
                value=""
              />,
              <ReminderOptions
                handleChange={[Function]}
                oneMonth={false}
                oneWeek={false}
                twoMonths={true}
                twoWeeks={true}
              />,
              <input
                className="jsx-2962072046 button-primary"
                data-test="purpose"
                type="submit"
                value="Submit"
              />,
              <FormAlert
                message=""
              />,
            ],
            "className": "jsx-2962072046",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add New Grant: ",
                "className": "jsx-2962072046",
              },
              "ref": null,
              "rendered": "Add New Grant: ",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="jsx-2962072046 u-full-width four columns"
                    name="name"
                    onChange={[Function]}
                    placeholder="Name of grant"
                    required={true}
                    type="text"
                    value="a grant name"
                  />,
                  <MonthOptions
                    handleChange={[Function]}
                    selected={2}
                  />,
                  <DayOptions
                    date={"2018-03-23T05:00:00.000Z"}
                    handleChange={[Function]}
                    selected={23}
                  />,
                  <YearOptions
                    handleChange={[Function]}
                    selected={2018}
                  />,
                  <PurposeOptions
                    handleChange={[Function]}
                    purposes={
                      Object {
                        "one": "General Operations",
                        "three": "Special Projects",
                        "two": "Education",
                      }
                    }
                    selected="general operations"
                  />,
                ],
                "className": "jsx-2962072046 row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "jsx-2962072046 u-full-width four columns",
                    "name": "name",
                    "onChange": [Function],
                    "placeholder": "Name of grant",
                    "required": true,
                    "type": "text",
                    "value": "a grant name",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "handleChange": [Function],
                    "selected": 2,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "date": "2018-03-23T05:00:00.000Z",
                    "handleChange": [Function],
                    "selected": 23,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "handleChange": [Function],
                    "selected": 2018,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "handleChange": [Function],
                    "purposes": Object {
                      "one": "General Operations",
                      "three": "Special Projects",
                      "two": "Education",
                    },
                    "selected": "general operations",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "jsx-2962072046 u-full-width",
                "name": "notes",
                "onChange": [Function],
                "placeholder": "Notes ...",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleChange": [Function],
                "oneMonth": false,
                "oneWeek": false,
                "twoMonths": true,
                "twoWeeks": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "jsx-2962072046 button-primary",
                "data-test": "purpose",
                "type": "submit",
                "value": "Submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "message": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": ".row.jsx-2962072046{margin:12px 0;}.options.jsx-2962072046{display:grid;grid-template-columns:repeat(auto-fill,minmax(100px,1fr));}textarea.jsx-2962072046{resize:vertical;}.options.jsx-2962072046 label.jsx-2962072046{margin:0;}.button-primary.jsx-2962072046{margin-top:8px;}",
            "styleId": "2962072046",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
